core/shared/src/main/scala/org/http4s/internal/Logger.scala

```diff
@@ -46,12 +47,16 @@ object Logger {
       val isJson = message.contentType.exists(mT =>
         mT.mediaType == MediaType.application.json || mT.mediaType.subType.endsWith("+json")
       )
-      val bodyStream = if (!isBinary || isJson) {
-        message.bodyText(implicitly, message.charset.getOrElse(Charset.`UTF-8`))
-      } else {
-        message.body.map(b => java.lang.Integer.toHexString(b & 0xff))
-      }
-      Some(bodyStream.compile.string)
+      val string =
+        if (!isBinary || isJson)
+          message
+            .bodyText(implicitly, message.charset.getOrElse(Charset.`UTF-8`))
+            .compile
+            .string
+        else
+          message.body.compile.to(ByteVector).map(_.toHex)
```

Comment:
This looks nicer, also perf within fluctuation remained the same. Thanks
```
StreamToByteVectorStringBuilding                                  100  thrpt   5  165.017 ± 6.876  ops/ms
StreamToByteVectorStringBuilding                                 1000  thrpt   5   95.626 ± 1.898  ops/ms
StreamToByteVectorStringBuilding                                10000  thrpt   5   17.401 ± 0.292  ops/ms
StreamToByteVectorStringBuilding:·gc.alloc.rate.norm              100  avgt    5   4919.750 ± 3.127  B/op
StreamToByteVectorStringBuilding:·gc.count                        100  avgt    5    129.000          counts
StreamToByteVectorStringBuilding:·gc.alloc.rate.nor              1000  avgt    5  10945.330 ± 0.059  B/op
StreamToByteVectorStringBuilding:·gc.count                       1000  avgt    5    271.000          counts
StreamToByteVectorStringBuilding:·gc.alloc.rate.norm            10000  avgt    5  98163.925 ± 0.238  B/op
StreamToByteVectorStringBuilding:·gc.count                      10000  avgt    5    448.000          counts
```

Reply:
Yes, I think it invokes that implicit `supportsByteVector` somehow. I believe this is the intended way to use the API.

We need a special-case for `fs2.Collector#string` because there is no `object String`, since it comes from Java.